File: code/Hyre_Defyer_Project_SE2/src/edu/westga/cs3211/hyre_defyer_project/model/Message.java
package edu.westga.cs3211.hyre_defyer_project.model;

/** Stores information within a message​
 *​
 * @author Alec Neal​
 * @version Spring 2025​
 */
public class Message {
	private String message;
	private User sender;
	private User receiver;
	
	/**
	 * Creates a message​ ​
	 * 
	 * @precondition none​
	 * @postcondition none
	 * 
	 * @param message The message
	 * @param sender  The sender
	 * @param receiver  The receiver​​
	 */
	public Message(String message, User sender, User receiver) {
		this.message = message;
        this.sender = sender;
        this.receiver = receiver;
    }
	
	/**
	 * Gets the message
	 * 
	 * @return the message
	 */
	public String getMessage() {
		return this.message;
	}
	
	/**
	 * Gets one of the users
	 * 
	 * @return the sender
	 */
	public User getSender() {
		return this.sender;
	}
	
	/**
	 * Gets one of the users
	 * 
	 * @return the sender
	 */
	public User getReceiver() {
		return this.receiver;
	}
	
	@Override
	public String toString() {
		return this.sender + ": " + this.message;
	}
}

File: code/Hyre_Defyer_Server_SE2/src/model/message.py
'''
    Stores information about a message.
    
    @author: alecx
    Version: Spring 2025
'''
from src.server import constants

class Message:
    
    def __init__(self, message, sender, receiver):
        '''
        Create a new message with the provided information.
        
        @precondition none
        @postcondition getMessage() == message && getSender() == sender && getReceiver() == receiver
        
        @param message: the message
        @param sender: the sender of the message
        @param receiver: the receiver of the message
        
        '''
        self._message = message
        self._sender = sender
        self._receiver = receiver

    def getMessage(self):
        '''
        Gets the message.

        @precondition none
        @postcondition none
        
        @return the message
        
        '''
        return self._message

    def getSender(self):
        '''
        Gets the sender of the message.

        @precondition none
        @postcondition none
        
        @return the sender
        '''
        return self._sender

    def getReceiver(self):
        '''
        Gets the receiver of the message.

        @precondition none
        @postcondition none
        
        @return the receiver
        '''
        return self._receiver
    
    def __eq__(self, other):
        if isinstance(other, Message):
            sender_match = (self._sender == other.getSender() and self._receiver == other.getReceiver()) or \
                       (self._sender == other.getReceiver() and self._receiver == other.getSender())
            return sender_match and self._message == other.getMessage()
        return False
    
    def to_dict(self):
        return {
            constants.REQ_SENDER: self._sender.getUserName(),
            constants.REQ_RECEIVER: self._receiver.getUserName(),
            constants.REQ_TEXT: self._message
        }